app_llm.py

```diff
@@ -0,0 +1,337 @@
+        st.rerun()
+
+    results = get_results(tally_sheet)
+
+    # ***************************************
+    result = results[0]
+
+    # Initialize from JSON result
+    # dataSet = result.get('dataSet', None)
+    # orgUnit = result.get('Health Structure', None)
+    # start_date = result.get('Start Date', None)
+    # end_date = result.get('End Date', None)
+    dataSet = None
+    orgUnit = None
+    start_date = None
+    end_date = None
+
+    # Initialize org_unit with any recognized text from tally sheet
+    # Change the value when user edits the field
+    if orgUnit:
+        org_unit = st.text_input("Organization Unit", value=orgUnit)
+    else:
+        org_unit = st.text_input("Organization Unit", placeholder="Search organisation unit name")
+
+    org_unit_dropdown = None
+    org_unit_options = None
+
+    # Get all UIDs corresponding to the text field value
+    if org_unit:
+        org_unit_options = dhis2_all_UIDs("organisationUnits", [org_unit])
+        org_unit_dropdown = st.selectbox(
+            "Searched Organizations",
+            [id[0] for id in org_unit_options],
+            index=None
+        )
+
+    # Get org unit children
+    if org_unit_dropdown is not None:
+        if org_unit_options:
+            org_unit_id = [id[1] for id in org_unit_options if id[0] == org_unit_dropdown][0]
+            org_unit_children_options = get_org_unit_children(org_unit_id)
+            org_unit_children_dropdown = st.selectbox(
+                "Organization Children",
+                sorted([id[0] for id in org_unit_children_options]),
+                index=None
+            )
+
+            if org_unit_children_dropdown is not None:
+
+                org_unit_child_id = [id[2] for id in org_unit_children_options if id[0] == org_unit_children_dropdown][0]
+                data_set_ids = [id[1] for id in org_unit_children_options if id[0] == org_unit_children_dropdown][0]
+                data_set_options = get_data_sets(data_set_ids)
+                data_set = st.selectbox(
+                    "Data Set",
+                    sorted([id[0] for id in data_set_options]),
+                    index=None
+                )
+
+                if data_set is not None:
+                    data_set_selected_id = [id[1] for id in data_set_options if id[0] == data_set][0]
+                    period_type = [id[2] for id in data_set_options if id[0] == data_set][0]
+                    st.write("Period Type\: " + period_type)
+
+    # Initialize with period values recognized from tally sheet or entered by user
+    if (start_date and end_date):
+        if start_date:
+            period_start = st.date_input("Period Start Date", format="YYYY-MM-DD", value=start_date)
+        else:
+            period_start = st.date_input("Period Start Date", format="YYYY-MM-DD")
+        if end_date:
+            period_end = st.date_input("Period End Date", format="YYYY-MM-DD", value=start_date)
```

Comment:
We can remove all the period_end statements as we only need the periodType and period_start to determine the time period.

Reply:
Thanks! Removed~